### PR TITLE
FitParameterLinked revision

### PR DESCRIPTION
**Pull request by gpospelov**
Now it is possible to link FitParameterLinked with multiple RealParameters using more than one wildcard based pattern.

* Additionally, FitSuite now reports possible conflicts in fit parameter setup, when one sample parameter is fitted by two different fit parameters.

### CODE REVIEW

File: Core/Fitting/FitParameterLinked.cpp [hunk 1]
```diff
@@ -135,3 +137,20 @@ FitParameterLinked::FitParameterLinked(const FitParameterLinked& other)
...
+            return true;
+
+        if(par->getName() != newPar.getName() && par->hasSameData(newPar))
+            throw std::runtime_error("FitParameterLinked::isLinked() -> This is confusing.");
+        if(par->getName() == newPar.getName() && !par->hasSameData(newPar))
+            throw std::runtime_error("FitParameterLinked::isLinked() -> This is confusing.");
+    }
+
+    return  false;
+}
```
Comment (waltervh) [inline Core/Fitting/FitParameterLinked.cpp]: Can you change this into a clear exception message instead of 'This is confusing'?

Comment (gpospelov) [inline Core/Fitting/FitParameterLinked.cpp]: In this case
* something really unexpected had happened
* explaining it in details would take too much wording, and probably useless 

I was hesitating between "Panic!", "Bug" and "Should not be here". Please suggest some universal formula.

May be "Unexpected behavior"?

Comment (waltervh) [inline Core/Fitting/FitParameterLinked.cpp]: What about:
"Different parameter with same data"
"Same parameter name with different data"

Comment (waltervh) [inline Core/Fitting/FitParameterLinked.cpp]: Shouldn't be long, just clear, in my opinion.

Comment (gpospelov) [inline Core/Fitting/FitParameterLinked.cpp]: ok

**Pull request merged**